src/cpp/desktop/DesktopSessionLauncher.cpp

```diff
@@ -51,13 +51,21 @@ void launchProcess(const std::string& absPath,
                    const QStringList& argList,
                    QProcess** ppProc)
 {
-   auto* pProcess = new QProcess();
+   QProcess* process = new QProcess();
+   process->setProgram(QString::fromStdString(absPath));
+   process->setArguments(argList);
    if (options().runDiagnostics())
-      pProcess->setProcessChannelMode(QProcess::ForwardedChannels);
-   else
-      pProcess->setProcessChannelMode(QProcess::SeparateChannels);
-   pProcess->start(QString::fromUtf8(absPath.c_str()), argList);
-   *ppProc = pProcess;
+      process->setProcessChannelMode(QProcess::ForwardedChannels);
+
+#ifdef _WIN32
+   process->setCreateProcessArgumentsModifier([](QProcess::CreateProcessArguments* cpa)
+   {
+      cpa->flags |= CREATE_NEW_PROCESS_GROUP;
```

Comment:
We should figure out how this interacts with the Jobs functionality we already have (which is used to ensure that when the rdesktop app is terminated the rsession and it's children go along with it). If this ends up "breaking out" of the job then it will lead to many undesirable orphaned processes.

Reply:
In doing some cursory scanning of documentation it looks like these mechanisms might be orthogonal so there is no issue with job objects. Here's where we create the job object: https://github.com/rstudio/rstudio/blob/8af730409bb6d651cc8f6816d136bea91441e7a4/src/cpp/core/system/Win32System.cpp#L71

It should be easy to test whether this is a problem by just using task manager to kill rdesktop and seeing whether rsession is also automatically terminated (may also want to do the same with rsession children).

Reply:
Some related SO discussion: https://stackoverflow.com/questions/1798771/send-ctrlc-to-subprocess-tree-on-windows

Reply:
I added the `core::system::getChildProcesses` method which walks the child processes for a given pid (regardless of their group ids), and there is an `rsession.conf` option `session-quit-child-processes-on-exit` which should kill all children on exit via `core::system::terminateChildProcesses`.

Reply:
@kfeinauer does this work for background processes? (E.g. if someone has run `system("sleep 100", wait = FALSE)` from R -- does that process also get cleaned up on exit?)

Reply:
Yes, it should. The methods scan the child processes recursively for a given pid, so even if it changes its process group it should still get killed.

Reply:
My point is that such processes will now have the `init` process as the parent, even though they still live in the same process group.